python/tskit/trees.py

```diff
@@ -1425,6 +1429,50 @@ def get_parent_dict(self):
     def __str__(self):
         return str(self.get_parent_dict())
 
+    def map_mutations(self, genotypes):
+        """
+        Given observations for a discrete set of characters at the samples in this
+        tree, reconstruct the state transitions required to reproduce these
+        observations. The observations are described by the genotypes array, with
+        distinct characters mapped to integers in the range 0 to 63,
+        such that ``genotypes[0]`` contains the observation for sample 0 and so on.
+        Missing data can be specified for a sample using the value
+        ``tskit.MISSING_DATA`` (-1). At least one non-missing observation must be
+        provided.
+
+        The current implementation uses the Fitch parsimony algorithm to determine
+        the minimum number of state transitions required to explain the data.
+
+        The state reconstruction is returned as two-tuple, ``(ancestral_state,
```

Comment:
Do we want to use the word "root state" here, rather than "ancestral state". The ancestral state might be though of for each node, rather than a general property of the tree (i.e. you could say there is an ancestral state in front of any transition on the tree, which could be different from the state at the root)

Reply:
It's a good point, but we're using the term ``ancestral_state`` quite specifically in the library to mean the state inherited by all roots. I've changed the sentance to read 
```
The state reconstruction is returned as two-tuple, ``(ancestral_state,
        transitions)``, where ancestral_state is the state that
        has been assigned to the tree root(s).
```

This will hopefully clear things up a bit. I don't want to change the name to ``root_state``, because the link with ``Site.ancestral_state`` is deliberate and meaninful.

Reply:
What happens if the tree has more than one root?